src/components/panel/panel.js

```diff
@@ -355,6 +355,45 @@ angular
  * @param {!MdPanelPosition} position
  */
 
+/**
+ * @ngdoc method
+ * @name MdPanelRef#registerListener
+ * @description
+ *
+ * Registers a listener with the panel. The callback should return a promise,
+ * which will allow the action to continue when it gets resolved, or will
+ * prevent an action if it is rejected. Currently supported listener types:
+ * * `onClose` - Gets called before the panel begins closing.
+ *
+ * @param {string} type Type of listener.
+ * @param {!angular.$q.Promise<!any>} callback Callback to be registered.
+ * @returns {!MdPanelRef}
+ */
+
+/**
+ * @ngdoc method
+ * @name MdPanelRef#removeListener
+ * @description
+ *
+ * Removes a registered listener.
+ *
+ * @param {string} type Type of listener to be removed.
+ * @param {Function} callback Listener to be removed.
```

Comment:
`{!Function}` or my preferred version because it says what the form of the function is: `function(): !angular.$q.Promise<any>`